src/transformers/modeling_utils.py

```diff
@@ -2781,7 +2781,12 @@ def from_pretrained(
             quantization_method_from_config = config.quantization_config.get(
                 "quant_method", QuantizationMethod.BITS_AND_BYTES
             )
-
+        if quantization_method_from_config is not None and quantization_method_from_args is not None:
+            if quantization_method_from_config != quantization_method_from_args:
+                raise ValueError(
+                    f"The model is already quantized with {quantization_method_from_config}. "
+                    f"You can't quantize it again with {quantization_method_from_args}"
+                )
```

Comment:
Why is it only raised when they're not equal? I would have expected that passing in both is equivalent to the user  trying to quantize again (even if would results in the same model)

The reason I ask is that it seems on L2796 there's a warning which explicitly states that the passed in config will be ignored anyway. Similarly on L2859.

Reply:
When they are equal, it will depends on the actual quantization scheme. For bitsandbytes, we overwrite the quantization config that was stored. For GPTQ, we overwrite only specific attributes such as (use_exllama, exllama_config, use_cuda_fp16).

Reply:
hmmm... OK. It's outside the scope of this PR but I suggest we rationalise this a bit - as a user I'd expect the behaviour to be consistent.

Reply:
Indeed, we might need to refactor the logic so that it is consistent.